pkg/distributor/distributor.go

```diff
@@ -175,8 +174,10 @@ func (cfg *Config) Validate(limits validation.Limits) error {
 		return errInvalidShardingStrategy
 	}
 
-	if cfg.ShardingStrategy == util.ShardingStrategyShuffle && limits.IngestionTenantShardSize <= 0 {
-		return errInvalidTenantShardSize
+	if limits.IngestionTenantShardSize <= 0 {
```

Comment:
There could be an scenario where:
1. Shuffle sharding is enabled for all tenants
2. Ingestion tenant shard is zero for all tenants
3. There is one specific tenant that has tenant shard bigger than zero
4. Shuffle sharding should be definitely enabled for that tenant. 

Is that supported with this change?

Reply:
I guess it does not. Let me check and make necessary changes

Reply:
Let's make sure it is still supported

Reply:
I checked the case you mentioned @friedrichg and made sure it is supported.

Reply:
We make sure things are supported adding tests. integration or unit test